main.py

```diff
@@ -31,9 +43,18 @@
 rand = random.Random()
 rand.seed(GLOBAL_SEED)
 new_seed = lambda: rand.randint(0, 1000_000)
-os.mkdir(SAVE_PREFIX)
+if not os.path.exists(SAVE_PREFIX):
+    os.mkdir(SAVE_PREFIX)
 
 torch.manual_seed(new_seed())
+
+cpu_num = cpu_count()
+os.environ ['OMP_NUM_THREADS'] = str(cpu_num)
+os.environ ['OPENBLAS_NUM_THREADS'] = str(cpu_num)
+os.environ ['MKL_NUM_THREADS'] = str(cpu_num)
+os.environ ['VECLIB_MAXIMUM_THREADS'] = str(cpu_num)
+os.environ ['NUMEXPR_NUM_THREADS'] = str(cpu_num)
```

Comment:
Maybe we don't need these environment variables here (seeming like torch's default behaviors, but not actually what we want)?